pkgs/servers/misc/lms/default.nix

```diff
@@ -0,0 +1,96 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, writeText
+, cmake
+, pkg-config
+, gtest
+, boost
+, zlib
+, ffmpeg_4
+, graphicsmagick
+, libconfig
+, taglib
+, wt, useMinimalWt ? false
+}:
+
+let
+  ffmpeg = ffmpeg_4;
+
+  # Wt with the minimum set of optional dependencies required for LMS
+  wt-minimal = wt.override {
+    # Docs not needed
+    doxygen = null;
+    # Graphics libraries not needed
+    glew = null; libharu = null; pango = null; graphicsmagick = null;
+    # LMS uses only sqlite
+    firebird = null; libmysqlclient = null; postgresql = null;
+    # Qt not used
+    qt48Full = null;
+  };
+
+  wt-lms = if useMinimalWt then wt-minimal else wt;
+in
+
+stdenv.mkDerivation rec {
+  pname = "lms";
+  version = "3.27.0";
+
+  src = fetchFromGitHub {
+    owner = "epoupon";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0rkky9c1yiqlgs9amrcnqsxdcwc7s7qghc1s1m0mbannsr1rb8rn";
+  };
+
+  nativeBuildInputs = [ cmake pkg-config gtest ];
+  buildInputs = [
+    boost.dev zlib.dev ffmpeg.dev graphicsmagick libconfig taglib wt-lms
+  ];
+
+  patches = [
+    (writeText "hardcode-dependency-paths.patch" ''
+      diff --git a/src/libs/av/impl/Transcoder.cpp b/src/libs/av/impl/Transcoder.cpp
+      index 7efa044..16a6e4c 100644
+      --- a/src/libs/av/impl/Transcoder.cpp
+      +++ b/src/libs/av/impl/Transcoder.cpp
+      @@ -38,7 +38,7 @@ static std::filesystem::path	ffmpegPath;
+       void
+       Transcoder::init()
+       {
+      -	ffmpegPath = Service<IConfig>::get()->getPath("ffmpeg-file", "/usr/bin/ffmpeg");
+      +	ffmpegPath = Service<IConfig>::get()->getPath("ffmpeg-file", "${ffmpeg.bin}/bin/ffmpeg");
+       	if (!std::filesystem::exists(ffmpegPath))
+       		throw Exception {"File '" + ffmpegPath.string() + "' does not exist!"};
+       }
```

Comment:
According to https://github.com/epoupon/lms/blob/835e705e09176797ce9f910640ac091c80b5f717/conf/lms.conf#L8 we could do this in the module instead:
```nix
services.lms.settings.ffmpeg-file = "${pkgs.ffmpeg_4.bin}/bin/ffmpeg";
```

Or is the issue that you want to be able to use this software without a configuration file?